### DISCUSSION TITLE
cache storage is substantially high

### DISCUSSION THREAD

**Discussion by rickysarraf** [photoprism/photoprism#1264] [Q&A] [answered]
I have only been a PhotoPrism user for around 7-10 days. So, in the initial days, much time was spent understanding the workflow and mechanics of PhotoPrism.

Now that I've kinda settled down on the setup, I was surprised by the storage consumption.

PhotoPrism itself uses:
```
rrs@lenovo:~/photoprism/.photo-prism-storage$ for x in *; do du -sh $x; done
3.2M    albums
25G     cache
16K     config
4.0K    serial
937M    sidecar
```

Whereas, my total Photo/Video folder itself is around `37 GiB`.

While setting up PhotoPrism, I did refine a few of my setup steps.
* Re-encoded many of the HEVC video files because no browser would want to decode them. And transcoding is expensive and slow
* Put some automated steps in place to reorganize backed up pictures from my mobile devices into certain locations.

Now, as I understand, PhotoPrism is smart enough to detect duplicates and stack them.

What I'm still wondering is is why is the `cache` that huge. Does PhotoPrism not do cache expiry/rotation ?

**Comment by lastzero** [comment, marked as answer]
The cache folder holds your thumbnails. When most of your originals are JPEGs already, the downscaled versions are not much smaller combined. You may reduce quality or render them on-demand if you have a powerful CPU and enough memory:

https://docs.photoprism.org/user-guide/settings/advanced/#images

**Reply by rickysarraf** [reply]
Can you please elaborate on the below behavior of PhotoPrism ?

```
photoprism_1  | time="2021-05-12T08:25:13Z" level=info msg="media: 7 thumbnails created for IMG_20210117_185338 [1.330189583s]"
photoprism_1  | time="2021-05-12T08:25:13Z" level=info msg="index: updated main jpg file Digikam_TO_BE_FILTERED/IMG_20210117_185338.jpg"
photoprism_1  | time="2021-05-12T08:25:13Z" level=info msg="index: added related xmp file IMG_20210117_185338.xmp"
photoprism_1  | time="2021-05-12T08:25:17Z" level=info msg="media: 7 thumbnails created for IMG_20210117_185340 [1.251398702s]"
photoprism_1  | time="2021-05-12T08:25:17Z" level=info msg="index: updated main jpg file Digikam_TO_BE_FILTERED/IMG_20210117_185340.jpg"
photoprism_1  | time="2021-05-12T08:25:17Z" level=info msg="index: added related xmp file IMG_20210117_185340.xmp"
photoprism_1  | time="2021-05-12T08:25:20Z" level=info msg="media: 7 thumbnails created for IMG_20210117_185353 [1.312167815s]"
photoprism_1  | time="2021-05-12T08:25:20Z" level=info msg="index: updated main jpg file Digikam_TO_BE_FILTERED/IMG_20210117_185353.jpg"
photoprism_1  | time="2021-05-12T08:25:20Z" level=info msg="index: added related xmp file IMG_20210117_185353.xmp"
photoprism_1  | time="2021-05-12T08:25:24Z" level=info msg="media: 7 thumbnails created for IMG_20210117_185356 [1.228650115s]"
photoprism_1  | time="2021-05-12T08:25:24Z" level=info msg="index: updated main jpg file Digikam_TO_BE_FILTERED/IMG_20210117_185356.jpg"
photoprism_1  | time="2021-05-12T08:25:24Z" level=info msg="index: added related xmp file IMG_20210117_185356.xmp"
photoprism_1  | time="2021-05-12T08:25:30Z" level=info msg="media: 7 thumbnails created for IMG_20210120_114621 [2.347628934s]"
photoprism_1  | time="2021-05-12T08:25:30Z" level=info msg="index: updated main jpg file Digikam_TO_BE_FILTERED/IMG_20210120_114621.jpg"
photoprism_1  | time="2021-05-12T08:25:30Z" level=info msg="index: added related xmp file IMG_20210120_114621.xmp"
photoprism_1  | time="2021-05-12T08:25:35Z" level=info msg="media: 7 thumbnails created for IMG_20210120_115239 [2.292402348s]"
photoprism_1  | time="2021-05-12T08:25:35Z" level=info msg="index: updated main jpg file Digikam_TO_BE_FILTERED/IMG_20210120_115239.jpg"
photoprism_1  | time="2021-05-12T08:25:35Z" level=info msg="index: added related xmp file IMG_20210120_115239.xmp"
photoprism_1  | time="2021-05-12T08:25:41Z" level=info msg="media: 7 thumbnails created for IMG_20210120_115329 [2.092755003s]"
photoprism_1  | time="2021-05-12T08:25:41Z" level=info msg="index: updated main jpg file Digikam_TO_BE_FILTERED/IMG_20210120_115329.jpg"
photo
```


I see the cache increasing on every scan even though not many new pictures are added.
For example:

```
rrs@lenovo:~/photoprism/.photo-prism-storage$ for x in *; do du -sh $x; done
3.2M    albums
31G     cache
16K     config
4.0K    serial
938M    sidecar
13:58 ♒♒♒   ☺ 😄    
```

6 GiB more than what I reported last week.

Does PhotoPrism not rotate the cache ? How should the above message be interpreted, "7 thumbnails created...." ? If literal English, why are 7 thumbnails created ?
There's also metadata being added to the JPEG files through Digikam, which PhotoPrism may sense as a change. But that shouldn't result in more and more thumbnails being created.

**Reply by lastzero** [reply]
You may run `photoprism cleanup` in a terminal to remove orphan thumbnails. PhotoPrism compares the file hash and doesn't know if the image itself or just metadata headers were changed. While we may implement this after reaching our funding goal, this is currently not the case.
